mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -1697,8 +1722,14 @@ impl TakerSwap {
             tx_hash,
         };
 
+        // Since taker payment is refunded successfully, maker payment can be released for lightning and similar protocols.
+        // # Important: New code that leads to refund failure shouldn't be added below this code block.
+        if let Err(e) = self.maker_coin.on_taker_payment_refund_success(&maker_payment).await {
+            error!("Error {} on calling on_taker_payment_refund!", e)
+        }
```

Comment:
Should we do the same if `self.taker_coin.is_auto_refundable()`?

Reply:
Yeah, looks like it's worth doing it. As these are commands, I would name them `PrepareForRefund` and `FinalizeRefund` or something similar 🙂

Reply:
`PrepareForRefund` and `FinalizeRefund` are good enough names, couldn't come up with better names myself :)